src/Automation.ts

```diff
@@ -105,4 +114,16 @@ export class Automation {
         throw new Error('Invalid trigger type')
     }
   }
+
+  isValidValidator(params: ClientParams) {
+    switch (params.validator) {
+      case ALLOWED_VALIDATORS.OWNABLE_VALIDATOR:
+      case ALLOWED_VALIDATORS.MOCK_VALIDATOR:
+        return true
+      case ALLOWED_VALIDATORS.SMART_SESSION_VALIDATOR:
+        return !!params.permissionId
+      default:
+        return notReachable(params.validator)
+    }
```

Comment:
do we need to enforce this here if this is already done in the backend?

Reply:
I think it is better to fail early and save the backend call

Reply:
ok the downside is we'll need to maintain allowed validators in both places but this shouldnt change much tbh